bundle/direct/dresources/dashboard.go

```diff
@@ -0,0 +1,306 @@
+		forceSendFields = append(forceSendFields, "EmbedCredentials")
+	}
+
+	return r.client.Lakeview.Publish(ctx, dashboards.PublishRequest{
+		DashboardId:      id,
+		EmbedCredentials: config.EmbedCredentials,
+		WarehouseId:      config.WarehouseId,
+		ForceSendFields:  forceSendFields,
+	})
+}
+
+func responseToState(createOrUpdateResp *dashboards.Dashboard, publishResp *dashboards.PublishedDashboard) *resources.DashboardConfig {
+	return &resources.DashboardConfig{
+		Dashboard: dashboards.Dashboard{
+			DisplayName:         createOrUpdateResp.DisplayName,
+			Etag:                createOrUpdateResp.Etag,
+			WarehouseId:         createOrUpdateResp.WarehouseId,
+			SerializedDashboard: createOrUpdateResp.SerializedDashboard,
+
+			// Add the /Workspace prefix to the parent path. The backend removes this prefix from parent
+			// path, and thus it needs to be added back in to match the local configuration.
+			// The default parent_path (i.e. ${workspace.resource_path}) includes the /Workspace prefix,
+			// that's why we need to add it back here.
+			ParentPath: path.Join("/Workspace", createOrUpdateResp.ParentPath),
+
+			// Output only fields
+			CreateTime:      createOrUpdateResp.CreateTime,
+			DashboardId:     createOrUpdateResp.DashboardId,
+			LifecycleState:  createOrUpdateResp.LifecycleState,
+			Path:            createOrUpdateResp.Path,
+			UpdateTime:      createOrUpdateResp.UpdateTime,
+			ForceSendFields: filterFields[dashboards.Dashboard](createOrUpdateResp.ForceSendFields),
+		},
+		SerializedDashboard: createOrUpdateResp.SerializedDashboard,
+		EmbedCredentials:    publishResp.EmbedCredentials,
+		ForceSendFields:     filterFields[dashboards.PublishedDashboard](publishResp.ForceSendFields),
+	}
+}
+
+func (r *ResourceDashboard) DoCreate(ctx context.Context, config *resources.DashboardConfig) (string, *resources.DashboardConfig, error) {
+	dashboard, err := prepareDashboardRequest(config)
+	if err != nil {
+		return "", nil, err
+	}
+
+	createResp, err := r.client.Lakeview.Create(ctx, dashboards.CreateDashboardRequest{
+		Dashboard: dashboard,
+	})
+
+	// If the parent directory doesn't exist, create it and try again.
+	if err != nil && isParentDoesntExistError(err) {
+		err = r.client.Workspace.MkdirsByPath(ctx, config.ParentPath)
+		if err != nil {
+			return "", nil, fmt.Errorf("failed to create parent directory: %w", err)
+		}
+		createResp, err = r.client.Lakeview.Create(ctx, dashboards.CreateDashboardRequest{Dashboard: dashboard})
+	}
+	if err != nil {
+		return "", nil, err
+	}
+
+	// Persist the etag in state.
+	config.Etag = createResp.Etag
+
+	publishResp, err := r.publishDashboard(ctx, createResp.DashboardId, config)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return createResp.DashboardId, responseToState(createResp, publishResp), nil
+}
+
+func (r *ResourceDashboard) DoUpdate(ctx context.Context, id string, config *resources.DashboardConfig) (*resources.DashboardConfig, error) {
```

Comment:
Should not DoUpdate handle Etag similar to DoCreate?

Reply:
Good catch! Missed that, fixed now.